### ISSUE TITLE
UI crashes when sending image from one to another with `tool="color-sketch"`

### ISSUE THREAD

**Issue by cvrunmin** [gradio-app/gradio#3473]
### Describe the bug

When an image is sent from one `gr.Image` to another `gr.Image` with `tool="color-sketch"`, the browser hangs and crashes the image.
Similar issue is found at #3248, with similar code here: https://github.com/gradio-app/gradio/issues/3248#issuecomment-1438313251 .

This issue has blocked usage of sending freshly generated image to img2img (sketch) in AUTOMATIC1111/stable-diffusion-webui#7007

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

You can reproduce using this snippet: 
```
with gr.Blocks() as demo:
    with gr.Row():
        with gr.Column():
            img1 = gr.Image(source='upload', label='img1', interactive=True, type="pil", tool="editor", image_mode="RGBA")
            btnSend = gr.Button('Send')
        with gr.Column():
            img2 = gr.Image(source='upload', label='img2', interactive=True, type="pil", tool="color-sketch", image_mode="RGBA")
    btnSend.click(fn=lambda x:x, inputs=[img1], outputs=[img2])

demo.launch()
```

### Logs

```shell
n/a (log is clean)
```

### System Info

```shell
Gradio: 3.21.0
Browser: Chrome, Edge
```

### Severity

serious, but I can work around it

**Comment by cvrunmin** [comment]
It seems to be the same canvas resizing issue I think

![image](https://user-images.githubusercontent.com/8142687/225360167-24f7e99c-b6e1-48c7-9d67-b850f0e28cb7.png)

**Comment by abidlabs** [comment]
Sg clearing from milestone

**Comment by vidit98** [comment]
I am also facing issue with sketch tool when trying to edit sketchpad based on users input. 
https://user-images.githubusercontent.com/22493755/227331730-bd81a547-fbea-472a-82eb-e1ea67afffff.mov

**Comment by pngwn** [comment]
Closed by #6169
